src/acquisition/covidcast/csv_to_database.py

```diff
@@ -32,107 +32,70 @@ def get_argument_parser():
-        database.rollback()
-
-    # archive the current file based on validation results
-    if all_rows_valid:
+def collect_files(data_dir, specific_issue_date, csv_importer_impl=CsvImporter):
+  """Fetch path and data profile details for each file to upload."""
+
+  if specific_issue_date:
+    results = list(csv_importer_impl.find_issue_specific_csv_files(data_dir))
+  else:
+    results = list(csv_importer_impl.find_csv_files(os.path.join(data_dir, 'receiving')))
+  print(f'found {len(results)} files')
+  return results
+
+def make_handlers(data_dir, specific_issue_date, file_archiver_impl=FileArchiver):
+  if specific_issue_date:
+    # issue-specific uploads are always one-offs, so we can leave all
+    # files in place without worrying about cleaning up
+    def handle_failed(path_src, filename, source):
+      print(f'leaving failed file alone - {source}')
+
+    def handle_successful(path_src, filename, source):
       print('archiving as successful')
       file_archiver_impl.archive_inplace(path_src, filename)
-    else:
-      print('unsuccessful - not archiving file')
+  else:
+    # normal automation runs require some shuffling to remove files
+    # from receiving and place them in the archive
+    archive_successful_dir = os.path.join(data_dir, 'archive', 'successful')
+    archive_failed_dir = os.path.join(data_dir, 'archive', 'failed')
+
+    # helper to archive a failed file without compression
+    def handle_failed(path_src, filename, source):
+      print('archiving as failed - '+source)
+      path_dst = os.path.join(archive_failed_dir, source)
+      compress = False
+      file_archiver_impl.archive_file(path_src, path_dst, filename, compress)
+
+    # helper to archive a successful file with compression
+    def handle_successful(path_src, filename, source):
+      print('archiving as successful')
+      path_dst = os.path.join(archive_successful_dir, source)
+      compress = True
+      file_archiver_impl.archive_file(path_src, path_dst, filename, compress)
+  return handle_successful, handle_failed
 
-# TODO: consider extending is_wip_override behavior option to this method
-def scan_upload_archive(
-    data_dir,
+def upload_archive(
+    path_details,
     database,
-    csv_importer_impl=CsvImporter,
-    file_archiver_impl=FileArchiver):
-  """Find CSVs, upload them to the database, and archive them.
+    handlers,
+    is_wip_override=None,
+    csv_importer_impl=CsvImporter):
+  """Upload CSVs to the database and archive them using the specified handlers.
 
-  data_dir: top-level directory where CSVs are stored
-  database: an open connection to the epidata database
-
-  The CSV storage layout is assumed to be as follows:
+  :path_details: output from CsvImporter.find*_csv_files 
+  
+  :database: an open connection to the epidata database
 
-  - Receiving: <data_dir>/receiving/<source name>/*.csv
-  - Archival: <data_dir>/archive/<status>/<source name>/*.csv[.gz]
```

Comment:
do we care about preserving this note about the directory structure?

Reply:
I thought about that, but it's in this file as well: https://github.com/cmu-delphi/delphi-epidata/blob/main/src/acquisition/covidcast/data_dir_readme.md#layout

& that gets dropped into /common/receiving